main/streams/userspace.c

```diff
@@ -510,12 +513,23 @@ PHP_FUNCTION(stream_wrapper_unregister)
 		RETURN_THROWS();
 	}
 
+	php_stream_wrapper *wrapper = zend_hash_find_ptr(php_stream_get_url_stream_wrappers_hash(), protocol);
 	if (php_unregister_url_stream_wrapper_volatile(protocol) == FAILURE) {
 		/* We failed */
 		php_error_docref(NULL, E_WARNING, "Unable to unregister protocol %s://", ZSTR_VAL(protocol));
 		RETURN_FALSE;
 	}
 
+	ZEND_ASSERT(wrapper != NULL);
+	if (wrapper->wops == &user_stream_wops) {
+		struct php_user_stream_wrapper *uwrap = (struct php_user_stream_wrapper *)wrapper;
+		zend_resource *resource = uwrap->resource;
+		uint32_t refcount = GC_DELREF(resource);
+		ZEND_ASSERT(refcount == 0);
```

Comment:
Uh, why are we guaranteed a zero refcount here?

Reply:
We aren't (see discussion in the PR). I'll change this to an if.